ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixParticipant.java

```diff
@@ -102,23 +115,25 @@ public synchronized boolean setReplicaSealedState(ReplicaId replicaId, boolean i
       throw new IllegalArgumentException(
           "HelixParticipant only works with the AmbryReplica implementation of ReplicaId");
     }
-    List<String> sealedReplicas = getSealedReplicas();
-    if (sealedReplicas == null) {
-      sealedReplicas = new ArrayList<>();
-    }
-    String partitionId = replicaId.getPartitionId().toPathString();
-    boolean success = true;
-    if (!isSealed && sealedReplicas.contains(partitionId)) {
-      logger.trace("Removing the partition {} from sealReplicas list", partitionId);
-      sealedReplicas.remove(partitionId);
-      success = setSealedReplicas(sealedReplicas);
-    } else if (isSealed && !sealedReplicas.contains(partitionId)) {
-      logger.trace("Adding the partition {} from sealReplicas list", partitionId);
-      sealedReplicas.add(partitionId);
-      success = setSealedReplicas(sealedReplicas);
+    synchronized (helixAdministrationLock) {
```

Comment:
Since you are synchronizing here, this method (`setReplicaSealedState()`) does not have to be `synchronized` anymore, does it?